src/cluster.cc

```diff
@@ -53,6 +55,52 @@ Status Cluster::SetNodeId(std::string node_id) {
   return Status::OK();
 }
 
+// Set the slot to the node if new version is current version +1. It is useful
+// when we scale cluster avoid too many big messages, since we only update one
+// slot distribution and there are 16384 slot in our design.
+//
+// The reason why the new version MUST be +1 of current version is that,
+// the command changes topology based on specific topology (also means specific
+// version), we must guarantee current topology is exactly expected, otherwise,
+// this update may make topology corrupt, so base topology version is very important.
+// This is different with CLUSTERX SETNODES commands because it uses new version
+// topology to cover current version, it allows kvrocks nodes lost some topology
+// updates since of network failure, it is state instead of operation.
+Status Cluster::SetSlot(int slot, std::string node_id, int64_t new_version) {
+  // Parameters check
+  if (new_version <= 0 || version != version_ + 1) {
+    return Status(Status::NotOK, "Invalid cluster version");
+  }
+  if (slot < 0 || slot >= kClusterSlots) {
+    return Status(Status::NotOK, "Invalid slot id");
+  }
+  if (node_id.size() != kClusetNodeIdLen)  {
+    return Status(Status::NotOK, "Invalid node id");
+  }
+
+  // Get the node which we want to assign a slot into it
+  std::shared_ptr<ClusterNode> to_assign_node = nodes_[node_id];
+  if (to_assign_node == nullptr) {
+    return Status(Status::NotOK, "No this node in the cluster");
+  }
+  if (to_assign_node->role_ != kClusterMaster) {
+    return Status(Status::NotOK, "The node is not the master");
+  }
+
+  // Update topology
+  //  1. Remove the slot from old node if existing
+  //  2. Add the slot into to-assign node
+  //  3. Update the map of slots to nodes.
+  std::shared_ptr<ClusterNode> old_node = slots_nodes_[slot];
+  if (old_node != nullptr) {
+    old_node->slots_[slot] = 0;
+  }
```

Comment:
Is it necessary to handle `old_node == to_assign_node` ?

Reply:
I ever thought, but i found we don't need since it don't change anything in current code, anything i miss?

Reply:
It's OK.